mongo/integration/unified_runner_events_helper_test.go

```diff
@@ -87,31 +87,30 @@ func waitForEvent(mt *mtest.T, test *testCase, op *operation) {
 	eventType := op.Arguments.Lookup("event").StringValue()
 	expectedCount := int(op.Arguments.Lookup("count").Int32())
 
-	callback := func(ctx context.Context) {
-		for {
-			// Stop loop if callback has been canceled.
-			select {
-			case <-ctx.Done():
-				return
-			default:
-			}
-
-			var count int
-			// Spec tests only ever wait for ServerMarkedUnknown SDAM events for the time being.
-			if eventType == "ServerMarkedUnknownEvent" {
-				count = test.monitor.getServerMarkedUnknownCount()
-			} else {
-				count = test.monitor.getPoolEventCount(eventType)
-			}
+	callback := func() bool {
+		// Stop loop if callback has been canceled.
+		select {
```

Comment:
I kept the select block for these two function because the UnifiedTestSpecs wouldn't pass on evergreen after removing.

Reply:
AFAICT this select block doesn't do anything, the argument is the same one here: https://github.com/mongodb/mongo-go-driver/pull/1668#discussion_r1633643051. Is this failure consistent in the CI? Can you recreate it locally?

Reply:
Never mind, I think the issue was when I changed a different part of the code. I retested it both locally and evergreen and it passed the test.